From 6ca677106992321326427c89a40e1c9673a499b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 8 Nov 2019 14:00:10 +0100
Subject: [PATCH] seccomp: add all *time64 syscalls
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/bionic/+source/systemd/+bug/1883447
Origin: upstream, https://github.com/systemd/systemd/commit/6ca677106992321326427c89a40e1c9673a499b2

From https://bugzilla.redhat.com/show_bug.cgi?id=1770154:
> utime is an obsolete system call. The current kernel interface is
> utimensat_time64. New 32-bit architectures do not even provide the utime
> system call.

Also add all other *time64 syscalls listed in
https://fedora.juszkiewicz.com.pl/syscalls.html.
---
 src/shared/seccomp-util.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

--- a/src/shared/seccomp-util.c
+++ b/src/shared/seccomp-util.c
@@ -276,12 +276,16 @@ const SyscallFilterSet syscall_filter_se
                 .help = "System calls that are always permitted",
                 .value =
                 "clock_getres\0"
+                "clock_getres_time64\0"
                 "clock_gettime\0"
+                "clock_gettime64\0"
                 "clock_nanosleep\0"
+                "clock_nanosleep_time64\0"
                 "execve\0"
                 "exit\0"
                 "exit_group\0"
                 "futex\0"
+                "futex_time64\0"
                 "get_robust_list\0"
                 "get_thread_area\0"
                 "getegid\0"
@@ -370,7 +374,9 @@ const SyscallFilterSet syscall_filter_se
                 .value =
                 "adjtimex\0"
                 "clock_adjtime\0"
+                "clock_adjtime64\0"
                 "clock_settime\0"
+                "clock_settime64\0"
                 "settimeofday\0"
                 "stime\0"
         },
@@ -480,6 +486,7 @@ const SyscallFilterSet syscall_filter_se
                 "unlinkat\0"
                 "utime\0"
                 "utimensat\0"
+                "utimensat_time64\0"
                 "utimes\0"
         },
         [SYSCALL_FILTER_SET_IO_EVENT] = {
@@ -498,7 +505,9 @@ const SyscallFilterSet syscall_filter_se
                 "eventfd2\0"
                 "poll\0"
                 "ppoll\0"
+                "ppoll_time64\0"
                 "pselect6\0"
+                "pselect6_time64\0"
                 "select\0"
         },
         [SYSCALL_FILTER_SET_IPC] = {
@@ -511,7 +520,9 @@ const SyscallFilterSet syscall_filter_se
                 "mq_notify\0"
                 "mq_open\0"
                 "mq_timedreceive\0"
+                "mq_timedreceive_time64\0"
                 "mq_timedsend\0"
+                "mq_timedsend_time64\0"
                 "mq_unlink\0"
                 "msgctl\0"
                 "msgget\0"
@@ -525,6 +536,7 @@ const SyscallFilterSet syscall_filter_se
                 "semget\0"
                 "semop\0"
                 "semtimedop\0"
+                "semtimedop_time64\0"
                 "shmat\0"
                 "shmctl\0"
                 "shmdt\0"
@@ -581,6 +593,7 @@ const SyscallFilterSet syscall_filter_se
                 "recv\0"
                 "recvfrom\0"
                 "recvmmsg\0"
+                "recvmmsg_time64\0"
                 "recvmsg\0"
                 "send\0"
                 "sendmmsg\0"
@@ -749,6 +762,7 @@ const SyscallFilterSet syscall_filter_se
                 "rt_sigprocmask\0"
                 "rt_sigsuspend\0"
                 "rt_sigtimedwait\0"
+                "rt_sigtimedwait_time64\0"
                 "sigaction\0"
                 "sigaltstack\0"
                 "signal\0"
@@ -788,10 +802,14 @@ const SyscallFilterSet syscall_filter_se
                 "timer_delete\0"
                 "timer_getoverrun\0"
                 "timer_gettime\0"
+                "timer_gettime64\0"
                 "timer_settime\0"
+                "timer_settime64\0"
                 "timerfd_create\0"
                 "timerfd_gettime\0"
+                "timerfd_gettime64\0"
                 "timerfd_settime\0"
+                "timerfd_settime64\0"
                 "times\0"
         },
 };
